### PR TITLE
Tokio 1.0

### PR DESCRIPTION
**Pull request by fakeshadow**
## PR Type

Update

## PR Checklist
Check your PR fulfills the following:

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

Update all cates in actix-web repo to tokio 1.0.

### CODE REVIEW

**Comment by yjhmelody** [comment]
Maybe it is better to wait for tokio 1.0?

**Comment by fakeshadow** [comment]
yea this PR is not meant to be the next release version.

**Comment by therealprof** [comment]
@fakeshadow What's the/your timeline for this? Need a hand?

**Comment by fakeshadow** [comment]
> @fakeshadow What's the/your timeline for this? Need a hand?

Thanks for asking. This is mostly done with only deps get in the way.
at least  `actix-service` `actix-server` need a new release to make it pass all CI. 
And after that I believe it's ready.

File: CHANGES.md [hunk 1]
```diff
@@ -3,9 +3,11 @@
 ## Unreleased - 2021-xx-xx
 ### Changed
 * Bumped `rand` to `0.8`
+* Update `rust-tls` to `0.19.0`
```
Comment (robjtede) [inline CHANGES.md]: reference PR

worth mentioning other upgrades in here too

File: actix-files/CHANGES.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased - 2021-xx-xx
 * `HttpRange::parse` now has its own error type.
+* Update `bytes` to `1`.
```
Comment (robjtede) [inline actix-files/CHANGES.md]: reference PR

File: actix-files/Cargo.toml [hunk 3]
```diff
@@ -18,11 +18,11 @@ path = "src/lib.rs"
...
 [dependencies]
 actix-web = { version = "3.0.0", default-features = false }
-actix-service = "1.0.6"
+actix-service = "2.0.0-beta.2"
 bitflags = "1"
-bytes = "0.5.3"
+bytes = "1"
 futures-core = { version = "0.3.7", default-features = false }
-futures-util = { version = "0.3.7", default-features = false }
+futures-util = { version = "0.3.7", default-features = false, features = ["sink"] }
```
Comment (robjtede) [inline actix-files/Cargo.toml]: appears to be needless feature

File: actix-http-test/CHANGES.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # Changes
 
 ## Unreleased - 2021-xx-xx
+* Update `bytes` to `1`.
```
Comment (robjtede) [inline actix-http-test/CHANGES.md]: reference PR

File: actix-http/CHANGES.md [hunk 5]
```diff
@@ -3,12 +3,20 @@
 ## Unreleased - 2021-xx-xx
 ### Changed
 * Bumped `rand` to `0.8`
+* Upgrade `bytes` to `1`
+* Upgrade `h2` to `0.3`
```
Comment (robjtede) [inline actix-http/CHANGES.md]: reference PR

File: actix-multipart/CHANGES.md [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased - 2021-xx-xx
 * Fix multipart consuming payload before header checks #1513
+* Update `bytes` to `1`
```
Comment (robjtede) [inline actix-multipart/CHANGES.md]: reference PR

File: actix-web-actors/CHANGES.md [hunk 7]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased - 2021-xx-xx
 * Upgrade `pin-project` to `1.0`.
+* Update `bytes` to `1`.
```
Comment (robjtede) [inline actix-web-actors/CHANGES.md]: reference PR

File: awc/CHANGES.md [hunk 8]
```diff
@@ -2,7 +2,9 @@
 
 ## Unreleased - 2021-xx-xx
 ### Changed
+* Upgrade `bytes` to `1`.
 * Bumped `rand` to `0.8`
+* Update `rust-tls` to `0.19.0`
```
Comment (robjtede) [inline awc/CHANGES.md]: reference PR

File: awc/src/ws.rs [hunk 9]
```diff
@@ -71,7 +71,6 @@ impl WebsocketsRequest {
     {
         let mut err = None;
 
-        #[allow(clippy::field_reassign_with_default)]
```
Comment (robjtede) [inline awc/src/ws.rs]: keep this to squelch warning

File: src/server.rs [hunk 10]
```diff
@@ -149,7 +150,7 @@ where
     /// This method should be called before `bind()` method call.
     pub fn backlog(mut self, backlog: i32) -> Self {
```
Comment (robjtede) [inline src/server.rs]: todo: use u32 here too

**Review by robjtede — CHANGES_REQUESTED** [review]
feels like there's some missing changelog items which are easily forgotten

**Review by robjtede — APPROVED** [review]

**Pull request merged**